src/display.c

```diff
@@ -113,6 +132,71 @@ static void DrawBorder(LCUI_PaintContext paint)
 	Graph_DrawHorizLine(&paint->canvas, color, 1, pos, end_x);
 }
 
+static size_t LCUIDisplay_UpdateFlashRects(SurfaceRecord record)
+{
+	int64_t period;
+	size_t count = 0;
+	LCUI_Graph mask;
+	LCUI_PaintContext paint;
+	FlashRect flash_rect;
+	LinkedListNode *node, *prev;
+
+	for (LinkedList_Each(node, &record->flash_rects)) {
+		flash_rect = node->data;
+		if (flash_rect->paint_time == 0) {
+			prev = node->prev;
+			free(node->data);
+			LinkedList_DeleteNode(&record->flash_rects, node);
+			node = prev;
+			continue;
+		}
+		period = LCUI_GetTimeDelta(flash_rect->paint_time);
+		if (period >= FLASH_DURATION) {
+			flash_rect->paint_time = 0;
+		} else {
+			Graph_Init(&mask);
+			mask.color_type = LCUI_COLOR_TYPE_ARGB;
+			Graph_Create(&mask, flash_rect->rect.width,
+				     flash_rect->rect.height);
+			Graph_FillRect(&mask, ARGB(50, 124, 179, 5), NULL, TRUE);
+			mask.opacity =
+			    0.6 * (FLASH_DURATION - (float)period) / FLASH_DURATION;
+		}
+		paint = Surface_BeginPaint(record->surface, &flash_rect->rect);
+		if (!paint) {
+			continue;
+		}
+		count += Widget_Render(record->widget, paint);
+		if (flash_rect->paint_time != 0) {
+			Graph_Mix(&paint->canvas, &mask, 0, 0, TRUE);
+			Graph_Free(&mask);
+			DrawBorder(paint);
```

Comment:
The border should render into the mask, you should modify the arguments and code of `DrawBorder()`.

Reply:
like this?

Reply:
No, like `DrawBorder(&mask)`

Reply:
done